app/src/main/java/com/example/tkey_android/FirstFragment.java

```diff
@@ -737,155 +881,390 @@ public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
+                                            activity.runOnUiThread(() -> TssClientHelper.showAlert(activity, "TransactionHash: " + ethSendTransaction.getTransactionHash()));
+                                        }
+                                    } catch (IOException e) {
+                                        hideLoading();
+                                        throw new RuntimeException(e);
+                                    }
+                                } else {
                                     hideLoading();
-                                    renderError(e);
+                                    activity.runOnUiThread(() -> Toast.makeText(activity, "Client is not ready, please try again", Toast.LENGTH_LONG).show());
                                 }
-
-                            });
-                            requireActivity().runOnUiThread(() -> {
-                                Snackbar snackbar = Snackbar.make(view1, "Phrase Deleted", Snackbar.LENGTH_LONG);
-                                snackbar.show();
+                            } else {
                                 hideLoading();
-                            });
-                        } else {
-                            requireActivity().runOnUiThread(() -> {
-                                Snackbar snackbar = Snackbar.make(view1, "Phrase failed ot be deleted", Snackbar.LENGTH_LONG);
-                                snackbar.show();
-                                hideLoading();
-                            });
+                                activity.runOnUiThread(() -> Toast.makeText(activity, "Client is not connected, please try again", Toast.LENGTH_LONG).show());
+                            }
+                        } catch (Exception | RuntimeError e) {
+                            hideLoading();
+                            throw new RuntimeException(e);
                         }
+                    } catch (IOException | ExecutionException | InterruptedException e) {
+                        hideLoading();
+                        throw new RuntimeException(e);
                     }
-                });
+                }).start();
             } catch (Exception e) {
-                renderError(e);
+                hideLoading();
+                throw new RuntimeException(e);
             }
         });
+    }
 
-        binding.exportShare.setOnClickListener(view1 -> activity.tKey.generateNewShare(result -> {
-            showLoading();
-            if (result instanceof Result.Error) {
-                requireActivity().runOnUiThread(() -> {
-                    Exception e = ((Result.Error<GenerateShareStoreResult>) result).exception;
-                    renderError(e);
-                    hideLoading();
-                });
-            } else if (result instanceof Result.Success) {
-                requireActivity().runOnUiThread(() -> {
+    private void sign(boolean isSigningMessage) {
```

Comment:
Parameter is unused.
```suggestion
    private void sign() {
```